Update pt_BR.po

marcelocripe opened the pull request:
Please accept the Brazilian Portuguese language translation for the Diffuse program.

The file that has the most recent translation can also be found at this URL:

https://github.com/marcelocripe/Diffuse_pt_BR/blob/main/diffuse_pt_BR.po.

I thank you for creating, maintaining and updating Diffuse.

Thank you very much.

marcelocripe
(Original text in Brazilian Portuguese language)

- - - - -

Por favor, aceite a tradução do idioma o Português do Brasil para o programa Diffuse.

O arquivo que possui a tradução mais recente também podem ser encontrado neste URL:

https://github.com/marcelocripe/Diffuse_pt_BR/blob/main/diffuse_pt_BR.po.

Eu agradeço por você criar, manter e atualizar o Diffuse.

Muito obrigado.

marcelocripe
(Texto original em idioma Português do Brasil)

MightyCreak commented on po/pt_BR.po:
```diff
@@ -25,30 +28,40 @@ msgstr ""
 #: data/io.github.mightycreak.Diffuse.appdata.xml.in:5
 msgid "Graphical tool for merging and comparing text files"
-msgstr "Ferramenta gráfica para unificar e comparar arquivos de texto"
+msgstr "O ‘Diffuse’ é uma ferramenta por interface gráfica "
+"que permite comparar e mesclar os arquivos de texto, "
+"comparar uma quantidade arbitrária de arquivos lado a "
+"lado e oferece aos usuários a capacidade de ajustar "
+"manualmente a correspondência das linha e editar os "
+"arquivos diretamente, possibilita recuperar as revisões "
+"feitas com as ferramentas ‘Bazaar’, ‘CVS’, ‘Darcs’, ‘Git’, "
+"‘Mercurial’, ‘Monotone’, ‘RCS’, ‘Subversion’ e ‘SVK’"
```
change: it is important that the string is about the same length. Here, the original string is used as the AppData summary and the .desktop comment. It is required to be a one-liner.

MightyCreak commented on po/pt_BR.po:
```diff
@@ -25,30 +28,40 @@ msgstr ""
 #: data/io.github.mightycreak.Diffuse.appdata.xml.in:4
 msgid "Diffuse"
-msgstr "Diffuse"
+msgstr "Comparador de Arquivos Diffuse"
```
change: this should just be "Diffuse" as it is the name of the app, and not a description. It's use as the name of the app in the .desktop file and the AppData. For instamce, if you look for GIMP in GNOME Software, you want to find the app named "GIMP" and not "GIMP the image manipulator" :wink:

MightyCreak commented on po/pt_BR.po:
```diff
@@ -25,30 +28,40 @@ msgstr ""
-"O Diffuse é uma ferramenta gráfica para unificar e comparar arquivos de texto. "
-"Também pode obter revisões para comparar arquivos de repositórios Bazaar, CVS, "
-"Darcs, Git, Mercurial, Monotone, RCS, e Subversion. "
+"O ‘Diffuse’ é uma ferramenta por interface gráfica que permite comparar e "
+"mesclar os arquivos de texto, comparar uma quantidade arbitrária de arquivos "
+"lado a lado e oferece aos usuários a capacidade de ajustar manualmente a "
+"correspondência das linha e editar os arquivos diretamente, possibilita "
+"recuperar as revisões feitas com as ferramentas ‘Bazaar’, ‘CVS’, ‘Darcs’, "
+"‘Git’, ‘Mercurial’, ‘Monotone’, ‘RCS’, ‘Subversion’ e ‘SVK’."
```
change: you seem to have copy/pasted the same paragraph :face_with_diagonal_mouth:

MightyCreak commented on po/pt_BR.po:
```diff
@@ -173,72 +187,79 @@ msgid ""
 "ability to manually adjust line matching and directly edit files. Diffuse can\n"
 "also retrieve revisions of files from several VCSs for comparison and merging."
 msgstr ""
+"O ‘Diffuse’ é uma ferramenta por interface gráfica que permite comparar e\n"
+"mesclar os arquivos de texto, comparar uma quantidade arbitrária de arquivos\n"
+"lado a lado e oferece aos usuários a capacidade de ajustar manualmente a\n"
+"correspondência das linha e editar os arquivos diretamente, possibilita\n"
+"recuperar as revisões feitas com as ferramentas ‘Bazaar’, ‘CVS’, ‘Darcs’,\n"
+"‘Git’, ‘Mercurial’, ‘Monotone’, ‘RCS’, ‘Subversion’ e ‘SVK’."
```
change: you seem to have copy/pasted the same paragraph here too :face_with_diagonal_mouth:

MightyCreak reviewed (changes requested):
I'm sorry I can't accept these changes as they are. Some translated texts are just the same copy/pasted paragraph which would mess with the UI as their length are totally different.

marcelocripe commented:
Hi MightyCreak.

First of all, I would like to thank you for creating, maintaining and updating the "Diffuse File Comparator" program.

I intend to respond to each of your comments and/or denials, but first I need to ask you a few questions.

Are you a Brazilian?
Are you a native speaker of Brazilian Portuguese?
Are you a programmer who respects the Brazilian Portuguese language and the way we communicate?
Are you a programmer who respects different cultures, languages ​​and the different ways people communicate?

If you answer these questions, I may continue writing long texts to try to explain and convince you of the reasons why I adapted the "pt_BR" translation exactly the way I did in the "pt_BR.po" file.

> I'm sorry I can't accept these changes as they are. Some translated texts are just the same copy/pasted
> paragraph which would mess with the UI as their length are totally different.
Are you sure that well-explained and detailed texts will harm the user interface? As soon as possible I will send you some screenshots of the "Diffuse File Comparator" program translated with the file that I made this merge request.

>change: this should just be "Diffuse" as it is the name of the app, and not a description. It's use as the
> name of the app in the .desktop file and the AppData. For instamce, if you look for GIMP in GNOME
> Software, you want to find the app named "GIMP" and not "GIMP the image manipulator" 
The menu text in Brazilian Portuguese for the entry "Name[pt_BR]=" is "Programa de manipulação de imagem do GNU", that is, the GIMP program is only understandable in my language because it has a descriptive text in the menu and is written in my language. Please check for yourself this page http://ftp.debian.org/debian/pool/main/g/gimp/ and the package "gimp_2.10.8-2_amd64.deb" or the "gimp_2.10.8-2_i386.deb ". The text "Programa de manipulação de imagem do GNU" should be written "Programa de Manipulação de Imagem do GNU", because these are the menus that are displayed in desktop environments or window managers, just like I I did it for the "Comparador de Arquivos Diffuse".
Do you really believe that the name "Diffuse" is known here in Brazil by around 215.3 million Brazilians (this number is from 2022)?
Which advertisement(s) were made for "Diffuse" here in Brazil?
Do you happen to know how many millions of dollars IBM or Microsoft spend to make their brands and products known in every corner of planet Earth?
I only came to know and want to translate, correct and improve the translation of "Diffuse" because I translated it on https://app.transifex.com/anticapitalista/antix-development/translate/#pt_BR/antix23-desktop-files, que foi para o https://gitlab.com/Robin-antiX/antix23-desktop-files/-/tree/main/applications?ref_type=heads/applications?ref_type=heads.
In other words, "Diffuse" benefited from my translation adaptations in the same way that I did in several GNU/Linux, antiX and MX Linux programs.

I leave you with a challenge here, to identify what these programs are or what these programs are for just by their name. But it's not worth searching on the internet. If you are a native English speaker then you have an advantage as the names are written in your language.

What are these programs below or what are they for?
zzzFM, antiX-cli-cc, cli-aptiX, CherryTree, Image Viewer, tupi-guarani, Szőke nő kék szemekkel, etc.

Which of these programs were you able to identify its function or purpose just by reading its name?

> change: you seem to have copy/pasted the same paragraph
This doesn't look like a copy, this is an exact copy of the text I produced to describe in detail what "Diffuse" is and what it is for.

Have you ever thought about describing the program better in the entries:

msgid "Graphical tool for merging and comparing text files"

It is

msgid ""
"Diffuse is a graphical tool for comparing and merging text files. It can retrieve "
"files for comparison from Bazaar, CVS, Darcs, Git, Mercurial, Monotone, RCS, and "
"Subversion repositories."

Please read these topics about why the FreeDesktop.org standards are not correct and still do not meet the basic principle of communication.

Suggestions to make antiX Linux more didactic/educational:
https://www.antixforum.com/forums/topic/suggestions-to-make-antix-linux-more-educational/

The explanations of the model that I use in "pt_BR" are also being used in the "de" language, as you can read and see in this topic:
https://www.antixforum.com/forums/topic/antix-23-2-full-runit-isos-for-testing/page/14/#post-148651

Also read the explanations that were made by a German (Robin) to a Hungarian (kempelen) and their examples of the importance of the user being able to understand what each program (or item) does or serves without having to search the internet about it. program.
https://www.antixforum.com/forums/topic/antix-23-2-full-runit-isos-for-testing/page/17/#post-149472

And here are some examples in "pt_BR" language of the menus I translated:
https://www.antixforum.com/forums/topic/antix-23-2-full-runit-isos-for-testing/page/14/#post-148749

And here, some more examples of how the FreeDesktop.org standards were designed by programmers and not by professionals in communication, sales, advertising, education, etc.
https://www.antixforum.com/forums/topic/genericnames-in-desktop-files-could-be-generic/

If you refuse my translations, you will be another programmer who does not respect my language, my culture, the way my people understand words in sentences and how we need texts so that ALL Brazilians are able to understand and manipulate the "Diffuse File Comparator" program.

marcelocripe
(Original text in Brazilian Portuguese language)

- - - - -

Olá, MightyCreak.

Antes de mais nada, eu registro os meus agradecimentos por você criar, manter e atualizar o programa "Comparador de Arquivos Diffuse".

Eu pretendo responder a cada um dos seus comentários e/ou recusas, mas antes eu preciso fazer algumas perguntas a você.

Você é um brasileiro?
Você é nativo do idioma português do Brasil?
Você é um programador que respeita o idioma português do Brasil e a forma como nos comunicamos?
Você é um programador que respeita as diferentes culturas, idiomas e as diferentes formas como as pessoas se comunicam?

Se você me responder estas perguntas, eu poderei insistir escrevendo longos textos para tentar te explicar e convencê-lo dos motivos pelo quais eu adaptei a tradução "pt_BR" exatamente da forma como eu fiz no arquivo "pt_BR.po".

> I'm sorry I can't accept these changes as they are. Some translated texts are just the same copy/pasted
> paragraph which would mess with the UI as their length are totally different.
Você tem certeza de que os textos bem explicados e detalhados irão prejudicar a interface do usuário? Assim que for possível eu irei te enviar algumas capturas de tela do programa "Comparador de Arquivos Diffuse" traduzido com o arquivo que eu realizei este pedido de mesclagem.

>change: this should just be "Diffuse" as it is the name of the app, and not a description. It's use as the
> name of the app in the .desktop file and the AppData. For instamce, if you look for GIMP in GNOME
> Software, you want to find the app named "GIMP" and not "GIMP the image manipulator" 
O texto do menu em idioma português do Brasil da entrada "Name[pt_BR]=" é "Programa de manipulação de imagem do GNU", ou seja, o programa GIMP só é compreensível em meu idioma porque possui um texto descritivo no menu e está escrito em meu idioma. Por favor, verifique você mesmo nesta página http://ftp.debian.org/debian/pool/main/g/gimp/ e no pacote "gimp_2.10.8-2_amd64.deb" ou no "gimp_2.10.8-2_i386.deb". O texto "Programa de manipulação de imagem do GNU" deveria estar escritos "Programa de Manipulação de Imagem do GNU", porque trata-se dos menus que são exibidos nos ambientes de área de trabalho ou nos gerenciadores de janelas, da mesma forma que eu fiz para o "Comparador de Arquivos Diffuse".
Você acredita mesmo que o nome "Diffuse" seja conhecido aqui no Brasil por cerca de 215,3 milhões brasileiros (este número é de 2022)?
Qual ou quais propagandas foram feitas do "Diffuse" aqui no Brasil?
Por acaso você sabe quantos milhões de dólares a IBM ou a Microsoft gastam para terem as suas marcas e os seus produtos conhecidos em todos os cantos do planeta Terra?
Eu só passei a conhecer e querer traduzir, corrigir e melhorar a tradução do "Diffuse" porque eu o traduzi no https://app.transifex.com/anticapitalista/antix-development/translate/#pt_BR/antix23-desktop-files, que foi para o https://gitlab.com/Robin-antiX/antix23-desktop-files/-/tree/main/applications?ref_type=heads.
Ou seja, o "Diffuse" foi beneficiado com as minhas adaptações das traduções da mesma forma que eu fiz em vários programas do GNU/Linux, do antiX, do MX Linux.

Eu deixo aqui um desafio para você, identificar o que é ou para que serve estes programas apenas pelo seu nome. Mas não vale pesquisar na internet. Se você é um nativo do idioma inglês, então você leva vantagem, já que os nomes estão escritos em seu idioma.

O que é ou para que serve estes programas abaixo?
zzzFM, antiX-cli-cc, cli-aptiX, CherryTree, Image Viewer, tupi-guarani, Szőke nő kék szemekkel, etc.

Qual destes programas você conseguiu identificar a sua função ou finalidade apenas lendo o seu nome?

> change: you seem to have copy/pasted the same paragraph
Isto não se parece com uma cópia, isto é uma cópia exata do texto que eu produzi para descrever com detalhes o que é e para que serve o "Diffuse".

Você já pensou em descrever melhor o programa nas entradas:

msgid "Graphical tool for merging and comparing text files"

e

msgid ""
"Diffuse is a graphical tool for comparing and merging text files. It can retrieve "
"files for comparison from Bazaar, CVS, Darcs, Git, Mercurial, Monotone, RCS, and "
"Subversion repositories."

Por favor, leia estes tópicos sobre os motivos pelos quais os padrões do FreeDesktop.org não estão corretos e ainda não atendem o princípio básico da comunicação.

Sugestões para tornar o antiX Linux mais didático/educacional:
https://www.antixforum.com/forums/topic/suggestions-to-make-antix-linux-more-educational/

As explicações do modelo que eu utilizo em "pt_BR" também está sendo utilizado no idioma "de", conforme você pode ler e ver neste tópico:
https://www.antixforum.com/forums/topic/antix-23-2-full-runit-isos-for-testing/page/14/#post-148651

Leia também as explicações que foram feitas por um Alemão (Robin) a um Húngaro (kempelen) e os seus exemplos da importância do usuário ser capaz de compreender o que cada programa (ou item) faz ou serve sem ter que pesquisar na internet sobre o programa.
https://www.antixforum.com/forums/topic/antix-23-2-full-runit-isos-for-testing/page/17/#post-149472

E aqui estão alguns exemplos em idioma "pt_BR" dos menus que eu traduzi:
https://www.antixforum.com/forums/topic/antix-23-2-full-runit-isos-for-testing/page/14/#post-148749

E aqui, mais alguns exemplos de como os padrões do FreeDesktop.org foram pensados por programadores e não por profissionais da comunicação, vendas, propaganda, educação, etc.
https://www.antixforum.com/forums/topic/genericnames-in-desktop-files-could-be-generic/

Se você recusar as minhas traduções, você será mais um programador que não respeita o meu idioma, a minha cultura, a forma como o meu povo compreende as palavras nas frases e como precisamos dos textos para que TODOS os brasileiros sejam capazes de compreender e manipular o programa "Comparador de Arquivos Diffuse".

marcelocripe
(Texto original em idioma Português do Brasil)

marcelocripe commented:
I'm back with the test results.

The file "Testes da Traducao pt_BR no Diffuse 29-07-2024.zip" contains the images of the tests I did on the translation "pt_BR.po" in the Diffuse program. The most recent ".po" and ".desktop" files are also inside the ".zip".

Thanks to these tests, I was able to notice that the letter "s" was missing in the text "correspondência das linha**s** e editar" in the ".po" and ".desktop" files. I have already corrected it in both files that are in the compressed ".zip" file.

In the image "Diffuse Sem a Traducao (1).png" you can see the menu with only the name of the program.

In the image "Diffuse Com a Traducao (1).png" you can see the menu with the translation and the name of the program. In the same way, you can see that the Geany and Vim text editors are properly translated into the "pt_BR" language.

The images "Diffuse Com a Traducao (1).png" to "Diffuse Com a Traducao (16).png" show all the translations that worked from the most recent ".po" file in the program version "0.7.7-1".

In the image "Diffuse Com a Traducao (11).png", notice that I inserted an extra line break to separate the texts.

In the image "Diffuse Com a Traducao (12).png", I removed the extra line break.

In the image "Diffuse Com a Traducao (13).png" to the image "Diffuse Com a Traducao (15).png", the translation of the most recent ".po" file did not work in the program version "0.7.7-1". If this is the text you told me would not fit in the graphical interface, I think the text will fit, if my line breaks are close to the English language.

Since I still don't know how to compile the programs, I can't test the translation of the language "pt_BR.po" in the latest program. All the translated texts that I could check were perfect and are understandable for most Brazilians.

Please reconsider and accept my translations.

- - - - -

Eu estou de volta com os resultados dos testes.

O arquivo "Testes da Traducao pt_BR no Diffuse 29-07-2024.zip" possui as imagens dos testes que eu fiz da tradução "pt_BR.po" no programa Diffuse, o arquivo mais recente ".po" e ".desktop" também estão dentro do ".zip".

Graças a estes testes, eu consegui perceber que faltava a letra "s" no texto "correspondência das linha**s** e editar" nos arquivos ".po" e no ".desktop". Eu já corrigi em ambos os arquivos que estão no arquivo compactado ".zip".

Na imagem "Diffuse Sem a Traducao (1).png" você pode observar o menu apenas com o nome do programa.

Na imagem "Diffuse Com a Traducao (1).png" você pode observar o menu com a tradução e o nome do programa. Da mesma forma como você pode observar os editores de texto Geany e Vim que estão devidamente traduzidos para o idioma "pt_BR".

As imagens "Diffuse Com a Traducao (1).png" a "Diffuse Com a Traducao (16).png" exibem todas as traduções que funcionaram do arquivo ".po" mais recente no programa da versão "0.7.7-1".

Na imagem "Diffuse Com a Traducao (11).png", observe que eu inseri uma quebra de linha a mais para separar os textos.

Na imagem "Diffuse Com a Traducao (12).png", eu removi a quebra de linha que estava a mais.

Na imagem  "Diffuse Com a Traducao (13).png" até a imagem "Diffuse Com a Traducao (15).png", a tradução do arquivo ".po" mais recente não funcionou no programa da versão "0.7.7-1". Caso seja este texto que você me disse que não iria caber na interface gráfica, eu acho que o texto cabe, se as minhas quebras de linhas estiverem próximas a do idioma Inglês.

Como eu ainda não sei como compilar os programas, eu não consigo testar a tradução do idioma "pt_BR.po" no programa mais recente. Todos os textos traduzidos que eu pude verificar ficaram perfeitos e são compreensíveis para a maioria dos brasileiros.

Por favor, reconsidere e aceite as minhas traduções.

[Testes da Traducao pt_BR no Diffuse 29-07-2024.zip](https://github.com/user-attachments/files/16419448/Testes.da.Traducao.pt_BR.no.Diffuse.29-07-2024.zip)

MightyCreak commented:
To answer your first questions:
> Are you a Brazilian?
> Are you a native speaker of Brazilian Portuguese?
> Are you a programmer who respects the Brazilian Portuguese language and the way we communicate?
> Are you a programmer who respects different cultures, languages ​​and the different ways people communicate?

No, no, yes, yes :wink:

marcelocripe commented:
Hello MightyCreak.

> PS: my mother tong is not English, so I have exactly the same issues with translations as you have.

So, it would be better if I sent you an automatic translation from the language "pt_BR" to your native language, instead of an automatic translation to the language "en".


> To answer your first questions:

Thank you for answering my questions. Now I know that this conversation and my merge request are worth pursuing.


> This sentence is also for data/io.github.mightycreak.Diffuse.appdata.xml which is used for the Flatkpak package and,
> according to Flathub, the original sentence is already too long and too complex:

This only reinforces what I wrote initially, that the "standards" are not concerned with ensuring full communication that makes it easier for GNU/Linux programs to be understood by most people and in different cultures.

What happens is that the ".pot" resource and the ".po" files of the Diffuse program do not have the entries for the programmer's comments "#." that serve to explain where each phrase will be used and the maximum number of characters that can be used, as for example, in the program https://gitlab.com/Robin-antiX/antixradio/-/blob/main/resources/translations/antiXradio.pot?ref_type=heads, onde as entradas "#. Text displayed in a button in main window.", "#. Window title supplement in main window's upper border, present if reception was stopped.", "#. Tooltip for main window's exit button. #. Never use a default exclamation mark in this text. If you need it, use the #. typographical replacement, e.g. alt+shift+u FF01 (Unicode full width exclamation mark)", etc. and in the program https://gitlab.com/Robin-antiX/antixscreenshot2/-/blob/main/po/antixscreenshot2.pot?ref_type=heads, where the programmer Robin even uses images from the program to exemplify where the sentences will be used, the maximum number of characters per line, whether a new paragraph will be started, how to add new paragraphs, etc.


> So, your text will actually make things even worse. As you can see, there are some constraints that are out of my control, so > I simply need to comply.

Regarding the number of characters to be used in the Flatkpak package description, I suggest separating the entries, making them exclusive to be used only for the Flatkpak package and separating the texts for creating the ".desktop" file, this way the texts become free without having to be "stuck" to the standards that limit communication between the programmer/translator and the end user.

Everything is under your control, what you need is to treat the inputs differently, where each input can be dedicated to different purposes. A piece of paper and programs have one thing in common, they accept receiving anything that humans want to write on them.

"data/io.github.mightycreak.Diffuse.appdata.xml", quais eu terei que reduzir o texto?
Several entries have this identifier "data/io.github.mightycreak.Diffuse.appdata.xml", which ones will I have to shorten the text?

Reducing texts means losing quality in information.


> Please, don't threaten me like that, and don't put yourself as the spearker person for all the Brazilians in this world.

This is not a threat, I didn't want it to sound this way in automatic translation. But know that it is the pure truth. It is sad for me, that I dedicate my time and volunteer work to receive rejections from programmers who prefer the translations to be poorly done, where several sentences are just automatic translations that have no verbal agreement or nominal agreement, with incomprehensible texts or even with several sentences of the program without translation for several years.


> I find that insulting, both to them and to me.

It was not my intention to insult you or anyone else.
Please know that antiX Linux is growing exponentially here in Brazil and for GNU/Linux to continue growing, we Brazilians need the programs translated in the best possible way. It is unacceptable for GNU/Linux to have people who act as if they were dictators and to have people who give orders and orders, preventing GNU/Linux from being available to everyone.


> I don't know you, and they probably don't know you either, so please allow me not to believe you when you say such
> things.

You are not here and you do not live the reality that I live. That is why you do not know about the huge number of illiterate people

https://www.gazetadopovo.com.br/vida-publica/brasil-tem-mais-de-50-milhoes-de-analfabetos-e-semi-analfabetos-anvj4qb4lnpjolm8wpc8dj68e/
https://oglobo.globo.com/brasil/educacao/noticia/2024/03/22/ibge-93-milhoes-de-brasileiros-ainda-sao-analfabetos-a-grande-maioria-com-mais-de-40-anos.ghtml

This sad reality only offends people who do not know their own condition.

We also suffer from another problem, the so-called "functional illiteracy", where people know how to read in their native language, but cannot understand what is written.

https://g1.globo.com/sc/santa-catarina/especial-publicitario/prefeitura-municipal-de-jaragua-do-sul/viver-jaragua/noticia/2021/11/12/analfabetismo-funcional-atinge-29percent-da-populacao-brasileira.ghtml

https://brasilescola.uol.com.br/gramatica/analfabetismo-funcional.htm

You can use Filipe PS's automatic translator (TWP - Translate Web Pages) in your internet browser to read these sad articles. Please be aware that these articles fail to convey the reality, which is even worse than the numbers.

> Once you'll earn my trust, things will probably get better, but I must be honest with you, it really doesn't go in the right
> direction right now.

People don't need to know me physically to benefit from my translations, you can learn a little more about my volunteer work in GNU/Linux on these pages:

https://gitlab.com/marcelocripe
https://github.com/marcelocripe/
https://archive.org/details/@marcelocripe
https://archive.org/details/@antixlinuxemportugues
https://www.antixforum.com/forums/topic/associacao-mutirao-cca-jd-fontalis/
https://www.antixforum.com/forums/topic/antix-and-antix-application-program-translation-platforms/
https://www.antixforum.com/forums/topic/canal-do-youtube-antix-linux-em-portugues/
https://www.youtube.com/@antixlinuxemportugues

I have accounts on other translation platforms that I didn't mention above.

Trust is something that takes time to build, trust is something that can easily be lost when words are misinterpreted (or in our case, when words are mistranslated by an automatic translator).

- - - - -

Olá, MightyCreak.

> PS: my mother tong is not English, so I have exactly the same issues with translations as you have.

Então, seria melhor eu enviar para você uma tradução automática do idioma "pt_BR" para o seu idioma natal, ao invés de uma tradução automática para o idioma "en".


> To answer your first questions:

Obrigado por responder as minhas perguntas. Agora eu sei que vale a pena insistir nesta conversa e no meu pedido de mesclagem.


> This sentence is also for data/io.github.mightycreak.Diffuse.appdata.xml which is used for the Flatkpak package and,
> according to Flathub, the original sentence is already too long and too complex:

Isso só reforça o que eu escrevi inicialmente, que os "padrões" não estão se preocupando em garantir uma comunicação plena que facilite os programas do GNU/Linux poderem ser compreendidos pela maioria das pessoas e em diferentes culturas.

O que acontece é que o recurso ".pot" e os arquivos ".po" do programa Diffuse não possuem as entradas dos comentários do programador "#." que servem para explicar onde cada frase será utilizada e a quantidade máxima de caracteres que pode ser utilizado, como por exemplo, no programa https://gitlab.com/Robin-antiX/antixradio/-/blob/main/resources/translations/antiXradio.pot?ref_type=heads, onde as entradas "#. Text displayed in a button in main window.", "#. Window title supplement in main window's upper border, present if reception was stopped.", "#. Tooltip for main window's exit button. #. Never use a default exclamation mark in this text. If you need it, use the #. typographical replacement, e.g. alt+shift+u FF01 (Unicode full width exclamation mark)", etc. e no programa https://gitlab.com/Robin-antiX/antixscreenshot2/-/blob/main/po/antixscreenshot2.pot?ref_type=heads, onde o programador Robin utiliza inclusive imagens do programa para exemplificar onde as frases serão utilizadas, a quantidade máxima de caracteres por linha, se um parágrafo novo será iniciado, como adicionar novos parágrafos, etc.


> So, your text will actually make things even worse. As you can see, there are some constraints that are out of my control, so > I simply need to comply.

Quanto a quantidade de caracteres para serem utilizados na descrição do pacote Flatkpak, eu sugiro separar as entradas tornando-as exclusivas para serem utilizadas apenas para o pacote Flatkpak e separar os textos para a criação do arquivo ".desktop", desta forma os textos passam a ser livres sem terem que ficar "presos" aos padrões que limitam a comunicação entre o programador/tradutor e o usuário final.

Tudo está sob o seu controle, o que precisa é tratar as entradas de forma diferente, onde cada entrada pode ser dedicada a objetivos diferentes. Um pedaço de papel e os programas tem um coisa em comum, eles aceitam receber qualquer coisa que os seres humanos queiram escrever neles.

Várias entradas possuem este identificador "data/io.github.mightycreak.Diffuse.appdata.xml", quais eu terei que reduzir o texto?

A redução dos textos significam perder a qualidade na informação.


> Please, don't threaten me like that, and don't put yourself as the spearker person for all the Brazilians in this world.

Isso não é uma ameaça, eu não queria que soasse desta maneira na tradução automática. Mas saiba que é a mais pura verdade. É algo triste para mim, eu dedicar o meu tempo e trabalho voluntário para receber recusas dos programadores que preferem que as traduções estejam mal feitas, onde várias frases são apenas traduções automáticas que não possuem concordância verbal ou concordância nominal, com textos incompreensíveis ou ainda com várias frases do programa sem tradução por vários anos.


> I find that insulting, both to them and to me.

Não era a minha intenção insultar você ou qualquer outra pessoa.
Saiba que o antiX Linux cresce de forma exponencial aqui no Brasil e para que o GNU/Linux continuar crescendo, nós brasileiros, precisamos dos programas traduzidos da melhor maneira possível. É inaceitável o GNU/Linux ter pessoas que agem como se fossem ditadores e ter pessoas que mandam e desmandam impedindo que o GNU/Linux seja para todas as pessoas.


> I don't know you, and they probably don't know you either, so please allow me not to believe you when you say such
> things.

Você não está aqui e não vive a realidade que eu vivo. Por isso não sabe da quantidade gigantesca de pessoas analfabetas

https://www.gazetadopovo.com.br/vida-publica/brasil-tem-mais-de-50-milhoes-de-analfabetos-e-semi-analfabetos-anvj4qb4lnpjolm8wpc8dj68e/
https://oglobo.globo.com/brasil/educacao/noticia/2024/03/22/ibge-93-milhoes-de-brasileiros-ainda-sao-analfabetos-a-grande-maioria-com-mais-de-40-anos.ghtml

A triste realidade só ofende as pessoas que não conhecem a sua própria condição.

Ainda sofremos com um outro problema, o chamado "analfabetismo funcional", onde as pessoas sabem ler em seu idioma natal, mas não conseguem compreender o que está escrito.

https://g1.globo.com/sc/santa-catarina/especial-publicitario/prefeitura-municipal-de-jaragua-do-sul/viver-jaragua/noticia/2021/11/12/analfabetismo-funcional-atinge-29percent-da-populacao-brasileira.ghtml

https://brasilescola.uol.com.br/gramatica/analfabetismo-funcional.htm

Você poderá utilizar o tradutor automático do Filipe PS (TWP - Translate Web Pages) no navegador de internet para ler estas tristes matérias. Saiba que estas matérias não conseguem transmitir a realidade qua é ainda pior que os números.

> Once you'll earn my trust, things will probably get better, but I must be honest with you, it really doesn't go in the right
> direction right now.

As pessoas não precisam me conhecer fisicamente para serem beneficiadas com as minhas traduções, você pode conhecer um pouco mais do meu trabalho voluntário no GNU/Linux nestas páginas:

https://gitlab.com/marcelocripe
https://github.com/marcelocripe/
https://archive.org/details/@marcelocripe
https://archive.org/details/@antixlinuxemportugues
https://www.antixforum.com/forums/topic/associacao-mutirao-cca-jd-fontalis/
https://www.antixforum.com/forums/topic/antix-and-antix-application-program-translation-platforms/
https://www.antixforum.com/forums/topic/canal-do-youtube-antix-linux-em-portugues/
https://www.youtube.com/@antixlinuxemportugues

Eu tenho conta em outras plataformas de tradução que eu não citei acima.

A confiança é algo que leva tempo para acontecer, a confiança é algo que fácil ser perdido quando as palavras são mal interpretadas (ou no nosso caso, quando as palavras são mal traduzidas por um tradutor automático).

MightyCreak commented:
> What happens is that the ".pot" resource and the ".po" files of the Diffuse program do not have the entries for the programmer's comments "#."

This is actually a very good advice! I'll try and improve that in the future! but it'll require a lot of additional comments in the source code to have it shown in the POT file, so don't expect to have all the translation lines done at once :wink:

MightyCreak commented:
> But know that it is the pure truth. It is sad for me, that I dedicate my time and volunteer work to receive rejections from programmers who prefer the translations to be poorly done, where several sentences are just automatic translations that have no verbal agreement or nominal agreement, with incomprehensible texts or even with several sentences of the program without translation for several years.

> Please know that antiX Linux is growing exponentially here in Brazil and for GNU/Linux to continue growing, we Brazilians need the programs translated in the best possible way. It is unacceptable for GNU/Linux to have people who act as if they were dictators and to have people who give orders and orders, preventing GNU/Linux from being available to everyone.

> You are not here and you do not live the reality that I live. That is why you do not know about the huge number of illiterate people


> This sad reality only offends people who do not know their own condition.

>We also suffer from another problem, the so-called "functional illiteracy", where people know how to read in their native language, but cannot understand what is written.

Come on dude, you really need to calm down. Diffuse is just a pet project I accepted to maintain in my spare time. I'm no dictator neither I'm here to rule the world with this program. Diffuse is just a graphical UI to show text differences, nothing revolutionary there.

I really appreciate that you're trying to improve Diffuse's Brazilian translation, I do, but please cool down a bit, we're not saving lives here. There are a multitude of associations and organizations that are especially meant for that, but this very specific GitHub repository is **definitely not** one of them.

So please, translate "Diffuse" by "Diffuse" when we're talking about the application name. It's just a moniker, I don't ask you to translate your identifier for each and every language in this world because it's unique.

Take [Boatswain](https://apps.gnome.org/Boatswain/) for instance (made by a Brazilian guy by the way): [the name of the app is "Boatswain"](https://gitlab.gnome.org/World/boatswain/-/blob/main/data/com.feaneron.Boatswain.desktop.in.in?ref_type=heads) and that's it! I don't even know what a Boatswain is (as English is not my mother tong), but I don't care! The app is great and works great!

marcelocripe commented:
Hello MightyCreak.

>This is actually a very good advice! I'll try and improve that in the future! but
> it'll require a lot of additional comments in the source code to have it shown > in the > POT file, so don't expect to have all the translation lines done at once 

Comments don't need to be added in the source code, because comments can only be added in the ".pot" file. The comment texts can contain details and the amount of text is free, as you can see in this example: https://gitlab.com/Robin-antiX/antixscreenshot2/-/blob/main/po/antixscreenshot2.pot?ref_type=heads.

- - - - -

Olá, MightyCreak.

>This is actually a very good advice! I'll try and improve that in the future! but
> it'll require a lot of additional comments in the source code to have it shown > in the > POT file, so don't expect to have all the translation lines done at once 

Os comentários não precisam ser adicionados no código fonte, porque os comentários podem ser adicionados apenas no arquivo ".pot". Os textos dos comentários podem conter detalhes e a quantidade de texto é livre, como vocÊ pode ver neste exemplo: https://gitlab.com/Robin-antiX/antixscreenshot2/-/blob/main/po/antixscreenshot2.pot?ref_type=heads.

MightyCreak commented:
It's apparently possible to do that: https://stackoverflow.com/a/51800316/1842511

I prefer to keep these comments close to the source code for two reasons:
1. It's easier for the devs to give context when it's right in the source code
3. It's easier to maintain as we don't have to sync the POT file with the source code manually

marcelocripe commented:
> So please, translate "Diffuse" by "Diffuse" when we're talking about the
> application name. It's just a moniker, I don't ask you to translate your
>  identifier for each and every language in this world because it's unique, and > that's perfectly fine!

This concept is wrong. Because people are not required to know what programs are for just by reading their name. I have never heard of this program "Boatswain". Nor is the "Comment" entry in the ".desktop" file, this is ridiculous and wrong. Unfortunately you are basing yourself on what is wrong, instead of what is appropriate for each language/culture.

I will give you a good example, the package http://security.debian.org/pool/updates/main/t/thunderbird/thunderbird_115.14.0-1%7edeb12u1_amd64.deb, descompacte o pacote ".deb", unpack the "data" file, enter the "/data/usr/share/applications" folder and see with your own eyes the "Name" entries all properly translated. 

> Name=Thunderbird
> Name[ast]=Veceru de corréu Thunderbird
> Name[ca]=Client de correu Thunderbird
> Name[cs]=Poštovní klient Thunderbird
> Name[da]=Mozilla Thunderbird - e-post/nyhedsgruppe
> Name[de]=Thunderbird E-Mail und Nachrichten
> Name[el]=Ηλεκτρονική αλληλογραφία Thunderbird
> Name[es]=Cliente de correo Thunderbird
> Name[fi]=Thunderbird-sähköposti
> Name[fr]=Messagerie Thunderbird
> Name[gl]=Cliente de correo Thunderbird
> Name[he]=Mozilla Thunderbird דוא״ל/חדשות
> Name[hr]=Mozilla Thunderbird e-pošta/novosti
> Name[hu]=Thunderbird levelezőkliens
> Name[it]=Email Mozilla Thunderbird
> Name[ja]=Thunderbird電子メールクライアント
> Name[ko]=Mozilla Thunderbird
> Name[nl]=Mozilla Thunderbird e-mail/nieuws
> Name[pl]=Klient poczty Thunderbird
> **Name[pt_BR]=Cliente de E-mail Thunderbird**
> Name[ru]=Почтовый клиент Thunderbird
> Name[sk]=Thunderbird - poštový klient
> Name[sv]=E-postklienten Thunderbird
> Name[ug]=Mozilla Thunderbird ئېلخەت/خەۋەر
> Name[uk]=Поштова програма Thunderbird
> Name[vi]=Trình điện thư Mozilla Thunderbird
> Name[zh_CN]=Thunderbird 邮件/新闻
> Name[zh_TW]=Thunderbird 郵件

Why rely on bad examples, when I have good examples that serve as a model for me to follow?

Since you gave a very simple answer, it is clear that you did not bother to read the topics that I linked in my previous comments. And with that, this merge request will join the list of developers who refused to accept translations and prefer not to accept how other cultures communicate. And once again, the losers are the GNU/Linux users who live in a "false freedom", since they cannot have their programs translated/adapted to their native languages.

- - - - -

> So please, translate "Diffuse" by "Diffuse" when we're talking about the
> application name. It's just a moniker, I don't ask you to translate your
>  identifier for each and every language in this world because it's unique, and > that's perfectly fine!

Este conceito está errado. Porque as pessoas não são obrigadas a saberem para que servem os programas apenas lendo o seu nome. Eu nunca ouvi falar deste programa "Boatswain". Nem a entrada "Comment" está no arquivo ".desktop", isto é ridículo e está errado. Infelizmente você se baseia no que está errado, ao invés do que é o adequado para cada idioma/cultura.

Eu vou citar um bom exemplo para você, o pacote http://security.debian.org/pool/updates/main/t/thunderbird/thunderbird_115.14.0-1%7edeb12u1_amd64.deb, descompacte o pacote ".deb", descompacte o arquivo "data", entre na pasta "/data/usr/share/applications" e veja com os seus próprios olhos as entradas "Name" todas devidamente traduzidas.

> Name=Thunderbird
> Name[ast]=Veceru de corréu Thunderbird
> Name[ca]=Client de correu Thunderbird
> Name[cs]=Poštovní klient Thunderbird
> Name[da]=Mozilla Thunderbird - e-post/nyhedsgruppe
> Name[de]=Thunderbird E-Mail und Nachrichten
> Name[el]=Ηλεκτρονική αλληλογραφία Thunderbird
> Name[es]=Cliente de correo Thunderbird
> Name[fi]=Thunderbird-sähköposti
> Name[fr]=Messagerie Thunderbird
> Name[gl]=Cliente de correo Thunderbird
> Name[he]=Mozilla Thunderbird דוא״ל/חדשות
> Name[hr]=Mozilla Thunderbird e-pošta/novosti
> Name[hu]=Thunderbird levelezőkliens
> Name[it]=Email Mozilla Thunderbird
> Name[ja]=Thunderbird電子メールクライアント
> Name[ko]=Mozilla Thunderbird
> Name[nl]=Mozilla Thunderbird e-mail/nieuws
> Name[pl]=Klient poczty Thunderbird
> **Name[pt_BR]=Cliente de E-mail Thunderbird**
> Name[ru]=Почтовый клиент Thunderbird
> Name[sk]=Thunderbird - poštový klient
> Name[sv]=E-postklienten Thunderbird
> Name[ug]=Mozilla Thunderbird ئېلخەت/خەۋەر
> Name[uk]=Поштова програма Thunderbird
> Name[vi]=Trình điện thư Mozilla Thunderbird
> Name[zh_CN]=Thunderbird 邮件/新闻
> Name[zh_TW]=Thunderbird 郵件

Para que se basear no mal exemplo, se eu tenho os bons exemplos que me servem de modelo a ser seguido?

Como você deu uma resposta muito simples, fica evidente que você não se deu o trabalho de ler os tópicos que eu vinculei nas minhas considerações anteriores. E com isso este pedido de mesclagem entrará para a lista dos desenvolvedores que se recusaram a receber as traduções e preferem não aceitar como as outras culturas de comunicam. E mais uma vez quem perde com isso são os usuários do GNU/Linux que vivem uma "falsa liberdade", já que não podem ter os programas traduzidos/adaptados aos sues idiomas natais.

MightyCreak commented:
Closing the PR: we don't understand eachother apparently.

marcelocripe commented:
> It's apparently possible to do that:
> https://stackoverflow.com/a/51800316/1842511

> I prefer to keep these comments close to the source code for two reasons:

> It's easier for the devs to give context when it's right in the source code
> It's easier to maintain as we don't have to sync the POT file with the source
> code manually

Unfortunately I don't have any programming knowledge and I can't say what is best from a programmers point of view.


> Closing the PR: we don't understand eachother apparently.

No. It is you who does not want to understand me. Anyone who reads all my arguments and the pages I linked to, if they are a native speaker of my language, will perfectly understand my good intentions.

By closing this merge request, you are only using your "power of choice" to accept or not my translation. Even though I try hard to make you understand what you do not want to understand.

This is also called dictatorship (or totalitarianism), where it is your opinion that counts and any opinion different from yours does not count.

- - - - -

> It's apparently possible to do that:
> https://stackoverflow.com/a/51800316/1842511

> I prefer to keep these comments close to the source code for two reasons:

> It's easier for the devs to give context when it's right in the source code
> It's easier to maintain as we don't have to sync the POT file with the source
> code manually

Infelizmente não possuo conhecimentos em programação e não posso dizer o que é o melhor do pondo de vista dos programadores.


> Closing the PR: we don't understand eachother apparently.

Não. É você que não quer me entender. Quem ler todos os meus argumentos e as páginas que eu vinculei, se for um nativo do meu idioma irá compreender perfeitamente as minhas boas intenções.

Ao fechar este pedido de mesclagem, você só está se utilizando do seu "poder de escolha" em aceitar ou não a minha tradução. Mesmo eu me esforçando em querer fazer você entender o que você não quer entender.

Isso também é chamado de ditadura (ou totalitarismo), onde é a sua opinião que que vale e qualquer opinião diferente da sua não vale.
